provider/resource_kpi_threshold_template.go

```diff
@@ -39,209 +44,431 @@ func kpiThresholdTemplateTFFormat(b *models.Base) (string, error) {
 	}
 
 	return cleanerRegex.ReplaceAllString(tpl.String(), ""), nil
-}
+}*/
 
 func kpiThresholdTemplateBase(clientConfig models.ClientConfig, key string, title string) *models.Base {
 	base := models.NewBase(clientConfig, key, title, "kpi_threshold_template")
 	return base
 }
 
-func ResourceKPIThresholdTemplate() *schema.Resource {
-	return &schema.Resource{
-		Description:   "Manages a KPI threshold template.",
-		CreateContext: kpiThresholdTemplateCreate,
-		ReadContext:   kpiThresholdTemplateRead,
-		UpdateContext: kpiThresholdTemplateUpdate,
-		DeleteContext: kpiThresholdTemplateDelete,
-		Importer: &schema.ResourceImporter{
-			StateContext: kpiThresholdTemplateImport,
+var (
+	_ resource.Resource                = &resourceKpiThresholdTemplate{}
+	_ resource.ResourceWithConfigure   = &resourceKpiThresholdTemplate{}
+	_ resource.ResourceWithImportState = &resourceKpiThresholdTemplate{}
+)
+
+func NewResourceKpiThresholdTemplate() resource.Resource {
+	return &resourceKpiThresholdTemplate{}
+}
+
+type modelKpiThresholdTemplate struct {
+	ID                                 types.String                            `tfsdk:"id"`
+	Title                              types.String                            `tfsdk:"title"`
+	Description                        types.String                            `tfsdk:"description"`
+	AdaptiveThresholdingTrainingWindow types.String                            `tfsdk:"adaptive_thresholding_training_window"`
+	TimeVariateThresholds              types.Bool                              `tfsdk:"time_variate_thresholds"`
+	TimeVariateThresholdsSpecification TimeVariateThresholdsSpecificationModel `tfsdk:"time_variate_thresholds_specification"`
+	AdaptiveThresholdsIsEnabled        types.Bool                              `tfsdk:"adaptive_thresholds_is_enabled"`
+	SecGrp                             types.String                            `tfsdk:"sec_grp"`
```

Comment:
not sure we save a lot of time by calling this field `sec_grp ` versus `security_group` and `security_group` seems simpler

Reply:
I have no opinion against changing this. But now conf files matches still match each other (but require re-import thought). 

With this change we will need to replace this attr in conf files

Reply:
I agree that `security_group` would be better, but we also have `sec_grp` field in the KPI base search resource (and maybe elsewhere). So if we are to fix this, might as well do it everywhere for consistency. And probably should do that in a separate PR.